ngraph/core/src/op/binary_convolution.cpp

```diff
@@ -73,80 +73,118 @@ op::v1::BinaryConvolution::BinaryConvolution(const Output<Node>& data,
 void op::v1::BinaryConvolution::validate_and_infer_types()
 {
```

Comment:
We have a lot of copy-pasted code from the regular convolution shape `validate_and_infer_types` function. It will be great if we can unify it somehow and move common code to a separate function.

Reply:
I can add this in a separate task (ticket: 48152) where I plan to refactor validate_and_infer_types method of convolution to be aligned with BinaryConvolution and proceed to move that code in a common function.

Reply:
@lazarevevgeny as part of #4387 check changes regarding your previous comment in commits:

- Add helper function for output shape inference of forward conv: [e1b38c2](https://github.com/openvinotoolkit/openvino/pull/4387/commits/e1b38c265c0ff79074d405eeccd1ad575e273b4e)
- Refactor `validate_and_infer_types` of Convolution to use helpers: [7a5f4a6](https://github.com/openvinotoolkit/openvino/pull/4387/commits/7a5f4a61f9af6e7b480b722ee4ba43e02d4d3f1a)
- Refactor `validate_and_infer_types` of BinaryConvolution to use helpers: [c8cb003](https://github.com/openvinotoolkit/openvino/pull/4387/commits/c8cb0031bac9a5a53da4b93a07f7ab4e0de58183)